day_9_Networking.md

```diff
@@ -0,0 +1,255 @@
+       valid_lft forever preferred_lft forever
+
+
+
+```
+2. You should have a possibility to use ssh client to connect to your node using new address from previous step. Run tcpdump in separate tmux session or separate connection before starting ssh client and capture packets that are related to this ssh connection. Find packets that are related to TCP session establish.
+```
+# Set up VBox - MyCentOs7 - Settings - Network - Port Forwarding - {Protocol: TCP, Host IP: 127.0.0.1, Host Port: 2222, Guest IP: 192.168.2.99, Guest Port: 22}
+# On host machine. Flags [S] - TCP session established:
+sudo tcpdump -i lo port 2222 > tcpdump.log
+
+# Connect to virtual machine in other console
+
+20:30:45.346595 IP localhost.51314 > localhost.2222: Flags [S], seq 771513756, win 65495, options [mss 65495,sackOK,TS val 844285421 ecr 0,nop,wscale 7], length 0
+20:30:45.346618 IP localhost.2222 > localhost.51314: Flags [S.], seq 2105504145, ack 771513757, win 65483, options [mss 65495,sackOK,TS val 844285421 ecr 844285421,nop,wscale 1], length 0
+
+# and tcpdump on virtual machine (it moves continuously):
+12:50:44.659187 IP 192.168.2.99.ssh > 10.0.2.2.51458: Flags [P.], seq 1872760:1873020, ack 1153, win 40880, length 260
+12:50:44.659247 IP 10.0.2.2.51458 > 192.168.2.99.ssh: Flags [.], ack 1873020, win 65535, length 0
+12:50:44.659359 IP 192.168.2.99.ssh > 10.0.2.2.51458: Flags [P.], seq 1873020:1873280, ack 1153, win 40880, length 260
+12:50:44.659454 IP 10.0.2.2.51458 > 192.168.2.99.ssh: Flags [.], ack 1873280, win 65535, length 0
+12:50:44.659553 IP 192.168.2.99.ssh > 10.0.2.2.51458: Flags [P.], seq 1873280:1873540, ack 1153, win 40880, length 260
+12:50:44.659612 IP 10.0.2.2.51458 > 192.168.2.99.ssh: Flags [.], ack 1873540, win 65535, length 0
+12:50:44.659712 IP 192.168.2.99.ssh > 10.0.2.2.51458: Flags [P.], seq 1873540:1873800, ack 1153, win 40880, length 260
+12:50:44.659783 IP 10.0.2.2.51458 > 192.168.2.99.ssh: Flags [.], ack 1873800, win 65535, length 0
+12:50:44.659881 IP 192.168.2.99.ssh > 10.0.2.2.51458: Flags [P.], seq 1873800:1874060, ack 1153, win 40880, length 260
+12:50:44.659941 IP 10.0.2.2.51458 > 192.168.2.99.ssh: Flags [.], ack 1874060, win 65535, length 0
+
+```
+3. Close session. Find in tcpdump output packets that are related to TCP session closure.
+```
+# Flags [F.] - Finish flag, indication of termination.
+20:43:38.976904 IP localhost.51434 > localhost.2222: Flags [F.], seq 3778, ack 3074, win 512, options [nop,nop,TS val 845059052 ecr 845059051], length 0
+20:43:38.992049 IP localhost.2222 > localhost.51434: Flags [F.], seq 3074, ack 3779, win 32742, options [nop,nop,TS val 845059067 ecr 845059052], length 0
+
+```
+4. run tcpdump and request any http site in separate session. Find HTTP request and answer packets with ASCII data in it.  Tcpdump command must be as strict as possible to capture only needed packages for this http request.
+```
+[mike@10 ~]$ sudo tcpdump -A host 10.0.2.15 and testingmcafeesites.com
```

Comment:
Should we specify protocol which packets we're trying to catch?

Reply:
```
[mike@10 ~]$ sudo tcpdump -A host 10.0.2.15 and testingmcafeesites.com and tcp
```
It's gonna be stricter)
Is it ok?